### ISSUE TITLE
Add `stylis` middleware support

### ISSUE THREAD

**Issue by nikvm** [zeit/styled-jsx#141]
Should we add [`stylis` middleware](https://github.com/thysultan/stylis.js#middleware) support via Babel plugin options that would be passed down to [`stylis.use`](https://github.com/thysultan/stylis.js#api)?

**Comment by giuseppeg** [comment]
I was wondering if it'd make sense to let users define a function to transform the CSS that is passed to Stylis instead. Like:

```js
const userTransform = state.file.opts.transform || (i) => i
transformedCss = transform(
    String(state.jsxId),
    userTransform(css)
)
```

This will in fact make `styled-jsx` extensible i.e. people could use SASS, I could drop styled-jsx-postcss etc.
Thoughts? cc @rauchg @nkzawa

**Comment by thysultan** [comment]
@giuseppeg BTW re: [171](https://github.com/zeit/styled-jsx/issues/171) stylis has nesting support out of the box.

**Comment by giuseppeg** [comment]
Fixed by #291
